sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/models/HttpLogOptions.java

```diff
@@ -0,0 +1,334 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.generic.core.http.models;
+
+import com.generic.core.implementation.http.policy.HttpRequestLogger;
+import com.generic.core.implementation.http.policy.HttpResponseLogger;
+import com.generic.core.util.configuration.Configuration;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+import static com.generic.core.util.configuration.Configuration.getGlobalConfiguration;
+
+/**
+ * The log configurations for HTTP messages.
+ */
+public final class HttpLogOptions {
+    private HttpLogDetailLevel logLevel;
+    private Set<String> allowedHeaderNames;
+    private Set<String> allowedQueryParamNames;
+    private boolean prettyPrintBody;
+    private HttpRequestLogger requestLogger;
+    private HttpResponseLogger responseLogger;
```

Comment:
For now, to keep the API simple, we should remove these and think about adding them back in the future at customer request. In addition, we should also remove request and response body logging.

Reply:
Follow up - https://github.com/Azure/azure-sdk-for-java/issues/37583